src/hera/workflows/script.py

```diff
@@ -198,12 +205,18 @@ def _build_script(self) -> _ModelScriptTemplate:
         assert isinstance(self.constructor, ScriptConstructor)
         image_pull_policy = self._build_image_pull_policy()
 
+        env = self._build_env()
+        if global_config.experimental_features["script_annotations"]:
```

Comment:
For this run-time side of things I think it makes sense to only add the env var for `RunnerScriptConstructor`s? Inline scripts won't have the function definition itself to derive from.

```suggestion
        if isinstance(self.constructor, RunnerScriptConstructor) and global_config.experimental_features["script_annotations"]:
```